runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -279,8 +282,7 @@ boolean atMaxWatermark(DataflowPipelineJob job, JobMetrics metrics) {
   JobMetrics getJobMetrics(DataflowPipelineJob job) {
     JobMetrics metrics = null;
     try {
-      metrics = options.getDataflowClient().projects().jobs()
-          .getMetrics(job.getProjectId(), job.getJobId()).execute();
+      metrics = dataflowClient.getJobMetrics(job.getJobId());
```

Comment:
Not all methods take jobId.
I feel it is not clear in the code to implicitly bound to an Nullable jobId.